### PR TITLE
Support YUV422->RGB565 conversion with libyuv

### PR DESCRIPTION
**Pull request by vigneshvg**
This was added in libyuv r1838 (commit 9b17af9b).

Also update ext/libyuv.cmd to that version.

### CODE REVIEW

File: ext/libyuv.cmd [hunk 1]
```diff
@@ -10,7 +10,7 @@
 git clone --single-branch https://chromium.googlesource.com/libyuv/libyuv
 
 cd libyuv
-git checkout d53f1be
+git checkout 9b17af9b
```
Comment (wantehchang) [inline ext/libyuv.cmd:13]: We should wait until this revision is used in Chrome. I can write a Chrome CL to update its third_party/libyuv to this revision.

The src/reformat_libyuv.c change in this pull request is fine. Would you like to remove the ext/libyuv.cmd change from this pull request?

Comment (wantehchang) [reply ext/libyuv.cmd:13]: See https://chromium-review.googlesource.com/c/chromium/src/+/3822634 for the libyuv update in Chrome.

Also please update the following item in CHANGELOG.md:
```
* Update libyuv.cmd: d53f1be (version 1837)
```

**Review by wantehchang — CHANGES_REQUESTED** [review]

File: ext/libyuv.cmd (hunk 1 above)
Comment (vigneshvg) [reply ext/libyuv.cmd]: I have removed the libyuv.cmd change from this PR. I will do it separately once it lands in chromium.

File: src/reformat_libyuv.c [hunk 2]
```diff
@@ -431,6 +431,24 @@ avifResult avifImageYUVToRGBLibYUV8bpc(const avifImage * image,
                 return AVIF_RESULT_REFORMAT_FAILED;
             }
             return AVIF_RESULT_OK;
+        } else if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV422) {
+#if LIBYUV_VERSION >= 1838
+            if (I422ToARGBMatrixFilter(image->yuvPlanes[AVIF_CHAN_Y],
```
Comment (wantehchang) [inline src/reformat_libyuv.c:436]: We should call `I422ToRGB565Matrix()`, right? I am sorry I didn't notice this error earlier.

Also test `AVIF_PIXEL_FORMAT_YUV422` before `AVIF_PIXEL_FORMAT_YUV420`, i.e., in the order of the enum defintion.

**Review by wantehchang — CHANGES_REQUESTED** [review]

File: src/reformat_libyuv.c (hunk 2 above)
Comment (vigneshvg) [reply src/reformat_libyuv.c]: Sorry it seems like i made a mess with the copy pasting on this CL.

I have updated it now and verified that it works. Please take another look.

**Review by wantehchang — APPROVED** [review]
LGTM.

**Pull request merged**